src/chart/barChartMultiSeries.js

```diff
@@ -29,36 +29,32 @@ export default function() {
 	let yScale;
 	let zScale;
 	let colorScale;
-
+	
 	/**
 	 * Initialise Data and Scales
 	 *
 	 * @private
 	 * @param {Array} data - Chart data.
 	 */
 	function init(data) {
-		const dataSummary = dataTransform(data).summary();
-		const categoryNames = dataSummary.rowKeys;
-		const seriesNames = dataSummary.columnKeys;
-		const maxValue = dataSummary.maxValue;
-
-		// If the colorScale has not been passed then attempt to calculate.
-		colorScale = (typeof colorScale === "undefined") ?
-			d3.scaleOrdinal().domain(seriesNames).range(colors) :
-			colorScale;
-
-		// Calculate Scales.
-		xScale = (typeof xScale === "undefined") ?
-			d3.scaleBand().domain(seriesNames).rangeRound([0, dimensions.x]).padding(0.5) :
-			xScale;
-
-		yScale = (typeof yScale === "undefined") ?
-			d3.scaleLinear().domain([0, maxValue]).range([0, dimensions.y]).nice() :
-			yScale;
-
-		zScale = (typeof zScale === "undefined") ?
-			d3.scaleBand().domain(categoryNames).range([0, dimensions.z]).padding(0.7) :
-			zScale;
+		const { rowKeys, columnKeys, maxValue } = dataTransform(data).summary();
+		const extent = [0, maxValue];
+	
+		if (typeof xScale === "undefined") {
+			xScale = d3.scaleBand().domain(columnKeys).rangeRound([0, dimensions.x]).padding(0.5);
+		}
+
```

Comment:
Is there a reason you changed these ternary operators into standard if statements?
I think I prefer them as they were. Happy discuss though.

Also please can we put back the deleted comments?

Reply:
Well, there are a few ways to represent "assign if not assigned, otherwise use previous value".

```js
// This is the worst way, I'd say. Don't do this.
(typeof xScale === "undefined") && xScale = d3.scaleBand().domain(columnKeys).rangeRound([0, dimensions.x]).padding(0.5);

// This is ok, but obfuscates intent.
// xScale should not need to be changed if it's already defined,
// and this does not communicate that.
xScale = (typeof xScale === "undefined")
    ?  d3.scaleBand().domain(columnKeys).rangeRound([0, dimensions.x]).padding(0.5)
    :  xScale;

// In my experience, this is the simplest, clearest, and most readable way.
// It's also probably more performant, as the assignment will simply be ignored.
if (typeof xScale === "undefined") {
    xScale = d3.scaleBand().domain(columnKeys).rangeRound([0, dimensions.x]).padding(0.5);
}
```

The reason an if statement is the best in this case, is that it does not assign a value unless the value is undefined. It also explains the intent much more clearly than a ternary.

If a ternay is used, the value will have a redundant assignment, as well as simply more code which is not needed for this. It's also the same number of lines, but less clear. Use a ternary when a value should not be assigned to itself in one of the outcomes.

As for the comments, a good rule of thumb is to have them only explain _why_, and not _what_ or _how_. Comments that exmplain what happens on the next line(s) tent to just be noise that needs to be maintained.
https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/
The exception being documentation comments, for obvious reasons.

Though if you still feel uncomfortable with me removing the comments, I will put them back.

Reply:
Thank you! Great explanation. I will go ahead and merge this in  as is!